src/ctlib/ct.c

```diff
@@ -3942,6 +3965,52 @@ ct_cursor(CS_COMMAND * cmd, CS_INT type, CS_CHAR * name, CS_INT namelen, CS_CHAR
 		cursor->status.close      = TDS_CURSOR_STATE_UNACTIONED;
 		cursor->status.dealloc    = TDS_CURSOR_STATE_UNACTIONED;
 
+		if (option == CS_UNUSED	 ||  (option & CS_END) != 0) {
+			/* Try to figure out type of the cursor. */
+			char delimiter[] = "\n\t,.[]() ";
+			enum {
+				eBaseline,
+				eFor,
+				eForUpdate
```

Comment:
camel case is a bit out of style in FreeTDS. What `e` stands for? What about `tok_baseline`, `tok_for`, `tok_for_update` ?